search-provider/eks-metadata-provider.c

```diff
@@ -0,0 +1,602 @@
+} ValueTranslationInfo;
+
+static ValueTranslationInfo *
+value_translation_info_new (AppendConstructionPropFromVariantWithTransformFunc append_func,
+                            gpointer                                           user_data,
+                            GDestroyNotify                                     user_data_destroy)
+{
+  ValueTranslationInfo *info = g_new0 (ValueTranslationInfo, 1);
+
+  info->append_func = append_func;
+  info->user_data = user_data;
+  info->user_data_destroy = user_data_destroy;
+
+  return info;
+}
+
+static void
+value_translation_info_free (ValueTranslationInfo *info)
+{
+  if (info == NULL)
+    return;
+
+  if (info->user_data_destroy)
+    g_clear_pointer (&info->user_data, info->user_data_destroy);
+
+  g_free (info);
+}
+
+static GHashTable *
+article_metadata_query_construction_props_translation_table (void)
+{
+  g_autoptr(GHashTable) table = g_hash_table_new_full (g_str_hash,
+                                                       g_str_equal,
+                                                       g_free,
+                                                       (GDestroyNotify) value_translation_info_free);
+
+  g_hash_table_insert (table,
+                       g_strdup ("search-terms"),
+                       value_translation_info_new (append_construction_prop_from_variant_dbus_transform,
+                                                   NULL,
+                                                   NULL));
+  g_hash_table_insert (table,
+                       g_strdup ("tags-match-any"),
+                       value_translation_info_new (append_construction_prop_from_variant_dbus_transform,
+                                                   NULL,
+                                                   NULL));
+  g_hash_table_insert (table,
+                       g_strdup ("tags-match-all"),
+                       value_translation_info_new (append_construction_prop_from_variant_dbus_transform,
+                                                   NULL,
+                                                   NULL));
+  g_hash_table_insert (table,
+                       g_strdup ("limit"),
+                       value_translation_info_new (append_construction_prop_from_variant_dbus_transform,
+                                                   NULL,
+                                                   NULL));
+  g_hash_table_insert (table,
+                       g_strdup ("offset"),
+                       value_translation_info_new (append_construction_prop_from_variant_dbus_transform,
+                                                   NULL,
+                                                   NULL));
+  g_hash_table_insert (table,
+                       g_strdup ("sort"),
+                       value_translation_info_new (append_construction_prop_from_variant_enum_transform,
+                                                   GUINT_TO_POINTER (EKNC_TYPE_QUERY_OBJECT_SORT),
```

Comment:
I don't think you can do this. I am not 100% sure, but I think GType is 64 bits and this will save only the lower 32 bits.

Reply:
Oh is it? Damn. Okay, I'll need to allocate then :(

Reply:
GType is defined as a `size_t` in the C case and a `ulong` in the C++ case:

```
#if     GLIB_SIZEOF_SIZE_T != GLIB_SIZEOF_LONG || !defined __cplusplus
typedef gsize                           GType;
#else   /* for historic reasons, C++ links against gulong GTypes */
typedef gulong                          GType;
#endif
```

Messy ... looks like it might be safer to just allocate.